stdlib/universe/join_test.go

```diff
@@ -720,6 +720,158 @@ func TestMergeJoin_Process(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "join with mismatched schemas",
+			spec: &universe.MergeJoinProcedureSpec{
+				On:         []string{"_time"},
+				TableNames: tableNames,
+			},
+			data0: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value", Type: flux.TFloat},
+						{Label: "key", Type: flux.TString},
+					},
+					KeyCols: []string{"key"},
+					Data: [][]interface{}{
+						{execute.Time(1), 1.0, "foo"},
+						{execute.Time(2), 2.0, "foo"},
+					},
+				},
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value", Type: flux.TFloat},
+					},
+					KeyCols: []string{},
+					Data: [][]interface{}{
+						{execute.Time(1), 1.5},
+						{execute.Time(2), 2.5},
+					},
+				},
+			},
+			data1: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value", Type: flux.TFloat},
+						{Label: "key", Type: flux.TString},
+					},
+					KeyCols: []string{"key"},
+					Data: [][]interface{}{
+						{execute.Time(1), 10.0, "bar"},
+						{execute.Time(2), 20.0, "bar"},
+					},
+				},
+			},
+			want: []*executetest.Table{
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value_a", Type: flux.TFloat},
+						{Label: "_value_b", Type: flux.TFloat},
+						{Label: "key_a", Type: flux.TString},
+						{Label: "key_b", Type: flux.TString},
+					},
+					KeyCols: []string{"key_a", "key_b"},
+					Data: [][]interface{}{
+						{execute.Time(1), 1.0, 10.0, "foo", "bar"},
+						{execute.Time(2), 2.0, 20.0, "foo", "bar"},
+					},
+				},
+				{
+					ColMeta: []flux.ColMeta{
+						{Label: "_time", Type: flux.TTime},
+						{Label: "_value_a", Type: flux.TFloat},
+						{Label: "_value_b", Type: flux.TFloat},
+						{Label: "key_a", Type: flux.TString},
```

Comment:
Why does this table have a `key_a` column? In the input table for `a` there as no `key` column. Is it because the first table in `a` had that column, and all the columns in the output of the join must have the same schema?

If that is the case then I would expect that `key_a` to also be in the group key, since the group key is sort of part of the schema too?